### DISCUSSION TITLE
SPI support to add WIZnet Ethernet

### DISCUSSION THREAD

**Discussion by VikWiz** [microsoft/devicescript#428] [General]
Hello,
I was wondering if there is any plan to add SPI support....
I wanted to test DeviceScript with RP2040 and WIZnet W5100S/W5500 chips...

**Comment by pelikhan** [comment]
@mmoskal is working on spi, (and images and a screen driver).

See #396, #423

**Comment by pelikhan** [comment]
The SPI support is now documented at https://microsoft.github.io/devicescript/developer/servers/spi . Make sure to get v2.9.15+.

**Reply by VikWiz** [reply]
Thanks for the info. Are there any guides how to add new libraries? I would like to port WIZnet ioLibrary and add EVB-Pico boards to list.

**Reply by pelikhan** [reply]
To publish a devicescript package under your npm org/account, https://microsoft.github.io/devicescript/developer/packages#publishing-a-devicescript-package

And here's the guide to create the board configuration for the EVB https://microsoft.github.io/devicescript/devices/add-board

**Reply by VikWiz** [reply]
great! thanks! would you mind to let me know when you expect to add SPI on RP2040?

**Reply by mmoskal** [reply]
hopefully a few weeks (unless we get a PR!)

**Comment by mmoskal** [comment]
Note that SPI is not yet implemented on rp2040. PRs are welcome!

**Reply by pelikhan** [reply]
right we need to document the status of implementation for various platforms
